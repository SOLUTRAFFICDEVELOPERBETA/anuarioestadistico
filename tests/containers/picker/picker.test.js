import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import ColorPicker from '../../../containers/picker';

describe('Prueba en <ColorPicker/>', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
        <ColorPicker
            color="#00000"
            palette="primary"
            onChange={onChange}

        />
    )
    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe llamar la funciÃ³n onChange()', () => {
        wrapper.find('WithStyles(ForwardRef(IconButton))').at(1).prop('onClick')();

        expect(onChange).toHaveBeenCalled();
        expect(onChange).toHaveBeenCalledTimes(1);
    })
})