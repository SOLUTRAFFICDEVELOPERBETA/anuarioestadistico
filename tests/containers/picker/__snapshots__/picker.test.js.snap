// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <ColorPicker/> debe mostrarse correctamente 1`] = `
<WithStyles(ForwardRef(ListItem))
  divider={true}
>
  <WithStyles(ForwardRef(ListItemText))
    primary="#00000"
  />
  <WithStyles(ForwardRef(ListItemSecondaryAction))>
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(SvgIcon))>
        <svg
          height="24"
          width="24"
        >
          <rect
            height="24"
            style={
              Object {
                "fill": undefined,
              }
            }
            width="24"
          />
        </svg>
      </WithStyles(ForwardRef(SvgIcon))>
    </WithStyles(ForwardRef(IconButton))>
    <WithStyles(ForwardRef(Popover))
      anchorEl={null}
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "bottom",
        }
      }
      id="color-popover-#00000"
      onClose={[Function]}
      open={false}
      transformOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
    >
      <div
        className="makeStyles-colorPalette-1"
      >
        <WithStyles(ForwardRef(IconButton))
          key="0"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#1976d2",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="1"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#0d47a1",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="2"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#2962ff",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="3"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#01579b",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="4"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#dc004e",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="5"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#b71c1c",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="6"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#f44336",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="7"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#bf360c",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="8"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#ff9800",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="9"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#ff6f00",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="10"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#2196f3",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="11"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#006064",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="12"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#4caf50",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="13"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#1b5e20",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="14"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#546e7a",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="15"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#455a64",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="16"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#263238",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          key="17"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(SvgIcon))>
            <svg
              height="200"
              width="200"
            >
              <rect
                height="200"
                style={
                  Object {
                    "fill": "#212121",
                  }
                }
                width="200"
              />
            </svg>
          </WithStyles(ForwardRef(SvgIcon))>
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </WithStyles(ForwardRef(Popover))>
  </WithStyles(ForwardRef(ListItemSecondaryAction))>
</WithStyles(ForwardRef(ListItem))>
`;
